src/project-manager.ts

```diff
@@ -483,6 +536,35 @@ export class ProjectConfiguration {
                 if (/(^|\/)jsconfig\.json$/.test(self.configFileName)) {
                     options.allowJs = true;
                 }
+
+                if (self.packageJsonFileName) {
+                    let parsedResult = JSON.parse(self.fs.readFile(self.packageJsonFileName));
+                    let devDeps = parsedResult.devDependencies;
+                    let deps = parsedResult.dependencies;
+
+                    if (deps) {
+                        for (let dep in deps) {
+                            let res = self.findDTSFile(dep);
+                            if (res && res.length > 0) {
+                                let depFile = res[0];
+                                self.fs.addFile(`node_modules/@types/${dep}/index.d.ts`, fs.readFileSync(res[0], 'utf8'));
+                                self.dtsNames[dep] = path_.basename(depFile);
+                            }
+                        }
+                    }
+
+                    if (devDeps) {
+                        for (let devDep in devDeps) {
+                            let res = self.findDTSFile(devDep);
+                            if (res && res.length > 0) {
+                                let depFile = res[0];
+                                self.fs.addFile(`node_modules/@types/${devDep}/index.d.ts`, fs.readFileSync(depFile, 'utf8'));
```

Comment:
Shouldn't we rater wait until TypeScript will ask our `LanguageServiceHost` for a file from `.../node_modules/@types/foo/name.d.ts` and only then we'll try to read local file and add it to in-memory FS under the name of `.../node_modules/@types/foo/name.d.ts`? This will solve the following:
- what if TypeScript will ask for `foo/bar.d.ts` instead of `foo/index.d.ts`?
- what if package.json is not located in the workspace root? (foo/bar/package.json)